crates/nu-cli/src/completions/completer.rs

```diff
@@ -395,7 +395,7 @@ impl NuCompleter {
                             Argument::Positional(_) if prefix == b"-" => flag_completion_helper(),
                             // complete according to expression type and command head
                             Argument::Positional(expr) => {
-                                let command_head = working_set.get_span_contents(call.head);
+                                let command_head = working_set.get_decl(call.decl_id).name();
```

Comment:
Perf wise this shouldn't be a hit `get_decl` is doing direct lookup compared to the loop of `get_span_contents`, the indirections of hitting `Command::name` through the vtable should be less impactful.